src/lib.rs

```diff
@@ -1,5 +1,102 @@
+use std::marker::PhantomData;
+
 #[cfg(target_os = "macos")]
-pub mod macos;
+mod macos;
+
+#[cfg(target_os = "android")]
+mod android;
+
+#[cfg(target_os = "windows")]
+mod windows;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Unsupported platform")]
+    Unsupported,
+    /// Permissions to access location / device are not allowed, and cannot be accessed until the
+    /// user manually intervenes.
+    #[error("Permissions denied")]
+    PermissionsDenied,
+    #[cfg(target_os = "windows")]
+    #[error("{0}")]
+    Windows(#[from] windows_result::Error),
```

Comment:
I think we might want to change up how we model platform specific errors, but for now this is fine. it would be nice to have a single permission error, but I can take care of this once I'm ready to integrate MacOS + Android